core/src/main/java/io/grpc/Deadline.java

```diff
@@ -139,4 +139,9 @@ public long timeRemaining(TimeUnit unit) {
     Preconditions.checkNotNull(scheduler, "scheduler");
     return scheduler.schedule(task, deadlineNanos - System.nanoTime(), TimeUnit.NANOSECONDS);
   }
+
+  @Override
+  public String toString() {
+    return String.valueOf(deadlineNanos);
```

Comment:
It would be nice if the units were printed out too.  We have a class to do this for timeouts in GrpcUtil that is almost what you want.